src/Sylius/Bundle/PromotionBundle/Form/Type/CatalogPromotionActionType.php

```diff
@@ -68,11 +68,8 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 $form = $event->getForm();
                 $formData = $form->getData();
 
-                if ($formData !== null) {
-                    $formData->setType($data['type']);
-                    $formData->setConfiguration($data['configuration']);
-
-                    $form->setData($formData);
+                if ($formData !== null && $formData?->getType() !== $data['type']) {
```

Comment:
```suggestion
                if ($formData !== null && $formData->getType() !== $data['type']) {
```
unnecessary since you're already checking for that